src/components/ButtonNav/index.tsx

```diff
@@ -1,36 +1,53 @@
 import React from 'react'
-import { StyledButton, Wrapper, StyledLink } from './styled'
+import { StyledButton, Wrapper, ButtonLink } from './styled'
 import { UserEdit, Users } from '@styled-icons/fa-solid'
 import { PermContactCalendar } from '@styled-icons/material-rounded'
+import { Theme, useThemeControl } from '../Theme'
 
 interface Props {
 	login: boolean
 }
 
-const NavBeforeLogin = (
+export const NavBeforeLogin = (theme: Theme): React.ReactElement => (
```

Comment:
```suggestion
const NavBeforeLogin: React.FC<{theme: Theme}> = ({theme}) => (
```
- For uniform syntax use React.FC and generics type props
- Export is unnecessary here

If reason of this syntax is education, so I understand :)

Reply:
Yes, this was show of two different styles how to handle something like this:
1. Make an indipendent component with own hook etc. (easy to move into a separate file)
2. Make a function that returns a component and pass everything as arguments of such function.

In this case I would say that both styles are interchangeable, but it's true, that we should probably stick with one as a convention. :wink:

Reply:
Fixed. :) I kept a function that returns a component and passes everything as an argument of itself. The reason being we usually keep components in separate files already so if they're in one, convention changes and it seems more "clean" to pass an argument.